src/track/taglib/trackmetadata_file.cpp

```diff
@@ -120,7 +120,8 @@ bool readAudioPropertiesFromFile(
     // we must pick one explicit,
     // to prevent "use of overloaded operator '<<' is ambiguous" error
     // on clang-cl builds.
-    const std::wstring& filename = file.name().wstr();
+    TagLib::FileName filename_owning = file.name();
+    const std::wstring& filename = filename_owning.wstr();
```

Comment:
Please add a comment. IMO this is a common enough pitfall that we might also want to consider raising a bug upstream.
```suggestion
    // We need to save the filename here because if it was chained 
    // (`file.name().wstr()`) the wstr() result would be dangling. 
    TagLib::FileName filename_owning = file.name();
    const std::wstring& filename = filename_owning.wstr();
```